File: Udemy Master Data Structure and Algorithm/6 - Array/Intro-Array.js
// Introduction Array

const array = ["a", "b", "c", "d", "e"];

// Every operation in array and it's Big O Notation

// Lookup - looking for the data
console.log(array[2]);

// Push - append data to the end of the array
array.push("f"); //O(1)
console.log(array);

// Pop - delete data at the end of the array
array.pop(); //O(1)
console.log(array);

// unshift - add data at the beginning of the array
array.unshift("0"); //O(n) => we need to loop to arrange the array again after insertion
console.log(array);

// splice
array.splice(2, 0, "hello"); // O(n) => we need to loop to arrange the array again after insertion
console.log(array);

File: Udemy Master Data Structure and Algorithm/6 - Array/Intro-Array.cpp
#include <iostream>
#include <string.h>
#include <vector>

using namespace std;

int main()
{
  vector<string> arr = {
      "a",
      "b",
      "c",
      "d",
      "e"};

  // Lookup
  cout << "Look Up the third array" << endl;
  cout << arr[2] << endl;

  // push
  arr.push_back("f");

  cout << "After Push Back" << endl;

  for (int i = 0; i < arr.size(); i++)
  {
    cout << arr[i];
  }

  // pop
  cout << endl;
  cout << "Pop Back" << endl;
  arr.pop_back();

  for (int i = 0; i < arr.size(); i++)
  {
    cout << arr[i];
  }

  // insert
  cout << endl;
  cout << "Insert at the beginning of array / unshift" << endl;
  auto it = arr.begin();

  it = arr.insert(it, "hello");

  for (int i = 0; i < arr.size(); i++)
  {
    cout << arr[i];
  }

  return 0;
}
